src/APICaller.js

```diff
@@ -0,0 +1,21 @@
+async function fetchWeatherData(url) {
+  const response = await fetch(url, {
+    mode: 'cors',
+  });
+  const responseToJson = await response.json();
```

Comment:
before doing `await response.json();` check if the response is actually application.json with `if ((res.headers.get('Content-Type') || '').includes('application/json'))`

Reply:
and if not, throw error? dar mačiau, kad galima promise rejectus gaudyt su try...catch(), tai jei response.json() rejectintų, būtų galima errorą susemt?

Reply:
Throw error, console error, kaip nori :D Svarbu, kad nebutu unhandled promise rejection